libuavcan/include/uavcan/time.hpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Types and utilities for working with time.
+ */
+/** @file */
+
+#ifndef UAVCAN_TIME_HPP_INCLUDED
+#define UAVCAN_TIME_HPP_INCLUDED
+
+#include "uavcan/uavcan.hpp"
+#include "uavcan/util/math.hpp"
+
+namespace uavcan
+{
+/**
+ * Protected base class for duration values.
+ *
+ * @tparam DURATIONTYPE  The datatype returned when retrieving durations from
+ *                       realizations of this base class. This type must be
+ *                       exactly 8 bytes in size and must support certain concepts
+ *                       implied by DurationBase.
+ */
+template <typename DURATIONTYPE>
```

Comment:
Bad naming (also against the C++ Core Guidelines), should be PascalCase.

Reply:
Yep. This is a good convention; having only C macros appear ALL_CAPS. I'll follow from now on.